### PR TITLE
Relayer arb e2e

### PR DESCRIPTION
**Pull request by compojoom**
Fixes the flaky e2e tests on arbitrum. In addition to that it makes it possible to run the tests locally on Mac M1

### CODE REVIEW

File: docker/arbitrum/arbitrum.sh [hunk 1]
```diff
@@ -26,6 +26,10 @@ patch_test_node_script() {
         sed -E "s#^(run=true)\$#\1\nforce_init=true#;
                 s#^mydir=\`dirname .0\`#mydir=${NITRO}#" \
             ${NITRO}/test-node.bash > ${TEST_NODE_SCRIPT}
+
+        # fix for flaky test https://github.com/OffchainLabs/nitro/issues/1706
+        sed -i'' -e 's/NITRO_NODE_VERSION=offchainlabs\/nitro-node:v2\.1\.0-beta\.1-03a2aea-dev/NITRO_NODE_VERSION=offchainlabs\/nitro-node:v2.1.0-beta.4-837e45e-dev/' "${TEST_NODE_SCRIPT}"
```
Comment (istankovic) [inline docker/arbitrum/arbitrum.sh:31]: Does this mean once they release 2.1.0, we can get rid of this line?

File: docker/scripts/common.sh [hunk 2]
```diff
@@ -71,6 +71,11 @@ e2e_test_relayer() {
     local privkey=$4
     local txhash=$5
     local relayer=${root}/relayer/relayer-node18-linux-x64
+
+    unamestr=$(uname)
+    if [[ "$unamestr" == 'Darwin' ]]; then
+       relayer=${root}/relayer/relayer-node18-macos-x64
+    fi
```
Comment (istankovic) [inline docker/scripts/common.sh:78]: Could you please make a new function, e.g. `get_relayer_binary` that returns that and use the function here as well in other places that need it:
```
docker/optimism/optimism.sh:25:    local relayer=${ROOT}/relayer/relayer-node18-linux-x64
docker/optimism/optimism.sh:79:    relayer=${ROOT}/relayer/relayer-node18-linux-x64
docker/polygon_zkevm/polygon_zkevm.sh:69:        relayer=${ROOT}/relayer/relayer-node18-linux-x64
docker/polygon_zkevm/polygon_zkevm.sh:71:        relayer=${ROOT}/relayer/relayer-node18-macos-x64
```
?

**Review by istankovic — CHANGES_REQUESTED** [review]
Looks ok, but there are still places to fix with respect to relayer binaries.

File: docker/arbitrum/arbitrum.sh (hunk 1 above)
Comment (compojoom) [reply docker/arbitrum/arbitrum.sh:31]: yes

**Review by istankovic — APPROVED** [review]

**Pull request merged**